### ISSUE TITLE
Allow import of DKIM-keys (with a warning that it is not from DNS)

### ISSUE THREAD

**Issue by JustMyGithub** [lieser/dkim_verifier#421]
Providers may change the selector and remove the old selectors public key from DNS, a few years ago I noticed that for gmail. It would be nice it it was possible to import keys to DKIM verifier in such a case, but there should be a warning that it was provided by the user.
(Yes, I am aware that DNS can be faked as well, but that is not as easy as importing a key.)

btw: It would also be nice to have a trustworthy archive for (retired) DKIM keys, but that another thing.

**Comment by dodmi** [comment]
Yes, the dialog is for (permanent) caching, not for importing. But for your personal e-mails, the key would be in the cache. So, verification will work after it's removed from DNS if you configure caching. And, this is also your personal trustworthy archive.

What's not working is the case, that a key is replaced in DNS with the same selector.

**Comment by lieser** [comment]
The caching feature dodmi mentioned can be I think uses as a workaround for most of what you want.

1. Enable caching of DKIM keys,
2. View a DKIM signed message to cache at least one key.
3. Edit one of the existing cached keys to the one you want to import.

Yes not a nice workflow. But I successfully use it for e.g. importing the DKIM key for the example inside the RFC.

A way to directly add a new key to the cached ones would be of course better, but I currently don't see enough use for most users for it to have a high priority for me.

> but there should be a warning that it was provided by the user.

What is your use case for wanting such a warning?
I currently don't see a use case for wanting to long term import a not trusted key. So even if importing would come, it would probably be handled like all other cached keys.
